.github/workflows/main.yml

```diff
@@ -0,0 +1,165 @@
+# Copyright 2021 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     https://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: A workflow for updating discovery artifacts
+# Controls when the action will run.
+
+on:
+  # Triggers the workflow on pull request events but only for the master branch
+  pull_request:
+    branches: [ master ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Update Discovery Artifacts PR
+    runs-on: ubuntu-latest
+    if: ${{github.event.pull_request.user.login == 'parthea' && github.event.pull_request.title == 'Add github action to update discovery artifacts'}}
+    steps:
+      - name: Check out working directory for changes
+        uses: actions/checkout@v2
+        with:
+          ref: refs/heads/master
+          path: ./branch
+
+      - name: Check out branch with scripts
+        uses: actions/checkout@v2
+        with:
+          ref: refs/heads/add-github-action-to-update-discovery-artifacts
+          path: ./branch_with_scripts
```

Comment:
I'm guessing this is just for testing the action out, and will be updated once you merge?

Reply:
Correct. I will submit a follow up PR to use the scripts in the default branch once this PR is merged.

Reply:
This work item will be tracked in #1228